src/lib/common/JsonHelper.cpp

```diff
@@ -0,0 +1,99 @@
+/*
+*
+* Copyright 2013 Telefonica Investigacion y Desarrollo, S.A.U
+*
+* This file is part of Orion Context Broker.
+*
+* Orion Context Broker is free software: you can redistribute it and/or
+* modify it under the terms
+* of the GNU Affero General Public License as
+* published by the Free Software Foundation, either version 3 of the
+* License, or (at your option) any later version.
+*
+* Orion Context Broker is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+* General Public License for more details.
+*
+* You should have received a copy of the GNU Affero General Public License
+* along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+*
+* For those usages not covered by this license please contact with
+* iot_support at tid dot es
+*
+* Author: Orion dev team
+*/
+#include "common/JsonHelper.h"
+
+std::string toJsonString(const std::string& input)
+{
+  std::ostringstream ss;
+  ss << '"';
+  for (std::string::const_iterator iter = input.begin(); iter != input.end(); ++iter) {
+        switch (*iter) {
+            case '\\': ss << "\\\\"; break;
+            case '"': ss << "\\\""; break;
+            case '/': ss << "\\/"; break;
+            case '\b': ss << "\\b"; break;
+            case '\f': ss << "\\f"; break;
+            case '\n': ss << "\\n"; break;
+            case '\r': ss << "\\r"; break;
+            case '\t': ss << "\\t"; break;
+            default: ss << *iter; break;
+      }
+  }
+  ss << '"';
+  return ss.str();
+}
```

Comment:
> The caller of this function supplies an allocated buffer for the output and it is the callers responsibility to make sure the out-buffer is large enough.

That means that the caller has to do a first pass on the `in` string in order to know `out` buffer size (I guess that counting the number of `\`), then allocate memory (and releasing after use) or relay in a buffer large enough to hold whatever output may be (which is segfault prone as sooner or later the initial assumption on size is not honored).

This is the kind of things that make "plain C" implementation harder (in the mantenability aspect I refer in a previous comment) compared with using  `ostringstream` 

(Don't interpreted this comment as a support for the C++ implementation, as I tool before I don't want to bias the decission ;) I just only want to illustrate that performance-mantenability-readibility is a multivariable trade off which solution is not necesarily maximizing performance as the cost of minimizing the others).